docs/source/configuration/authorization.mdx

```diff
@@ -408,6 +416,192 @@ The response retains the initial request's shape but returns `null` for unauthor
 
 If _every_ requested field requires authentication and a request is unauthenticated, the router generates an error indicating that the query is unauthorized.
 
+<MinVersion version="1.34.0">
+
+### `@policy`
+
+</MinVersion>
+
+The `@policy` directive marks fields and types as restricted based on authorization policies evaluated in a [Rhai script](../customizations/rhai/) or [coprocessor](../customizations/coprocessor). This enables custom authorization validation beyond authentication and scopes.
+
+The `@policy` directive includes a `policies` argument that defines an array of the required policies. The following example shows a policy that requires all roles from the claims object to contain the string `"support"`:
+
+```graphql
+@policy(policies: [["claims[`roles`].contains(`support`)"]])
+```
+
+At the [`RouterService` level](../customizations/overview#the-request-lifecycle), the Apollo Router extracts the list of policies relevant to a request from the schema and then stores them in the request's context in `apollo_authorization::policies::required` as a map `policy -> null|true|false`.
```

Comment:
Is mentioning and linking to `RouterService` useful for the reader? I find it a bit confusing, since I don't think _I_ need to do anything at the `RouterService` level, only at the `SupergraphService` level.

If we're trying to place it in the lifecycle, maybe we say that this occurs _after_ any `RouterService` plugins?

Reply:
it's important to indicate what happens when, otherwise people get confused when trying to do what they want at the wrong step

Reply:
Would be a good idea to include a visual aid here. Doesn't have to be done in the first pass

Reply:
Adding as follow-up task